cachi2/core/package_managers/gomod.py

```diff
@@ -787,13 +788,13 @@ def _setup_go_toolchain(go_mod_file: RootedPath) -> Go:
             ),
         )
 
-    if target_version >= go_max_version:
+    if target_version >= go_121_version:
         # Project makes use of Go >=1.21:
         # - always use the 'X.Y.0' toolchain to make sure GOTOOLCHAIN=auto fetches anything newer
         # - container environments need to have it pre-installed
         # - local environments will always install 1.21.0 SDK and then pull any newer toolchain
         go = Go(release="go1.21.0")
-    elif go_base_version >= go_max_version:
+    elif go_base_version >= go_121_version:
```

Comment:
I also noticed something weird in the logs when running with `--log-level debug`:

```
2024-06-26 13:59:42,508 INFO go.mod reported versions: '1.22.0'[go], '-'[toolchain]
2024-06-26 13:59:42,508 DEBUG Trying to locate Go toolchain at '/usr/local/go/go1.21.0/bin/go': FAIL
2024-06-26 13:59:42,508 DEBUG Trying to locate Go toolchain at '/home/acmiel/.cache/cachi2/go/go1.21.0/bin/go': SUCCESS
```
cachi2 says it detected 1.22.0 but runs with go1.21.0

probably not related to this diff, just bringing it up here

Reply:
https://github.com/containerbuildsystem/cachi2/issues/550